source/fundamentals/indexes.txt

```diff
@@ -210,28 +185,117 @@ an array value. You can define a multikey index using the same syntax as a singl
+sort behavior. To learn more about multikey indexes, including a discussion of their behavior and limitations,
+see the :manual:`Multikey Indexes page </core/index-multikey>` in the MongoDB manual.
+
+.. _search-indexes:
+
+Atlas Search Indexes
+~~~~~~~~~~~~~~~~~~~~
+
+The Atlas Search feature enables you to perform full-text searches on
+collections hosted on MongoDB Atlas. The indexes specify the behavior of
+the search and which fields to index.
+
+To learn more about MongoDB Atlas Search, see the
+:atlas:`Atlas Search Indexes </atlas-search/atlas-search-overview/#fts-indexes>`
+documentation.
+
+You can call the following methods on a collection to manage your Atlas Search
+indexes:
+
+- ``createSearchIndex()``
+- ``createSearchIndexes()``
+- ``listSearchIndexes()``
+- ``updateSearchIndex()``
+- ``dropSearchIndex()``
+
+The following sections provide code examples that demonstrate how to use
+each of the preceding methods.
+
+Create a Search Index
++++++++++++++++++++++
+
+You can use the ``createSearchIndex()`` (TODO: add API link) and the
+``createSearchIndexes()`` (TODO: add API link) methods to create Atlas Search indexes.
+
+The following code example shows how to create a single index:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/SearchIndexMethods.java
+   :language: java
+   :dedent:
+   :start-after: start create-search-index
+   :end-before: end create-search-index
+
+The following code example shows how to create multiple indexes:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/SearchIndexMethods.java
+   :language: java
+   :dedent:
+   :start-after: start create-search-indexes
+   :end-before: end create-search-indexes
+
+List Search Indexes
++++++++++++++++++++
+
+You can use the
+``listSearchIndexes()`` (TODO: add API link) method to return the Atlas
+Search indexes of a collection.
+
+The following code example shows how to print a list of the search indexes of
+a collection:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/SearchIndexMethods.java
+   :language: java
+   :dedent:
+   :start-after: start list-search-indexes
+   :end-before: end list-search-indexes
+
+Update a Search Index
++++++++++++++++++++++
+
+You can use the
+``updateSearchIndex()`` (TODO: add API link)
+method to update an Atlas Search index.
+
+The following code shows how to update a search index:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/SearchIndexMethods.java
+   :language: java
+   :dedent:
+   :start-after: start update-search-index
+   :end-before: end update-search-index
+
+Drop a Search Index
++++++++++++++++++++
+
+You can use the
+``dropSearchIndex()`` (TODO: add API link)
+method to remove an Atlas Search index.
+
+The following code shows how to delete a search index from a collection:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/SearchIndexMethods.java
+   :language: java
+   :dedent:
+   :start-after: start drop-search-index
+   :end-before: end drop-search-index
 
```

Comment:
The search index commands operate asynchronously, returning from the server before the index is confirmed to be successfully updated, created, or dropped. To determine when an index has been created or updated, users should run the `listSearchIndexes` command periodically until the desired changes appear. It might be beneficial to include this information in the documentation as well. Let me know what you think